src/App.tsx

```diff
@@ -1,127 +1,152 @@
 import Buttons, { Button } from './components/Buttons'
 import { MouseEvent, useState } from 'react'
-import { arithmeticFunc, square } from './utils/helpers'
+import { arithmeticFunc, square, power, equal } from './utils/helpers'
 
 import styles from './App.module.css'
 
 const ARITHMETIC_ARR = ['/', '-', '*', '+']
+const NUMBER_ARR = ['1', '2', '3', '4', '5', '6', '7', '8', '9', '0']
 
 function App() {
-  const [input, setInput] = useState<string | null>('')
+	const [input, setInput] = useState<string | null>('')
 
-  const buttons: Button[] = [
-    {
-      content: 'x²',
-      func: () => {
-        if (square(ARITHMETIC_ARR, input)) setInput(input + '**2')
-      },
-    },
-    {
-      content: 'xʸ',
-      // func: power,
-    },
-    {
-      content: '←',
-      func: () => {
-        if (input === null) return
-        const inputVar = input.slice(0, -1)
-        setInput(inputVar)
-      },
-    },
-    {
-      content: '()',
-      // func: parentheses,
-    },
-    {
-      content: '1',
-    },
-    {
-      content: '2',
-    },
-    {
-      content: '3',
-    },
-    {
-      content: '*',
-    },
-    {
-      content: '4',
-    },
-    {
-      content: '5',
-    },
-    {
-      content: '6',
-    },
-    {
-      content: '/',
-    },
-    {
-      content: '7',
-    },
-    {
-      content: '8',
-    },
-    {
-      content: '9',
-    },
-    {
-      content: '+',
-    },
-    {
-      content: '+/-',
-      // func: negate,
-    },
-    {
-      content: '0',
-    },
-    {
-      content: '.',
-    },
-    {
-      content: '-',
-    },
-    {
-      content: '=',
-      func: () => {
-        setInput(input => {
-          try {
-            return Function('return ' + input)()
-          } catch (error) {
-            return error
-          }
-        })
-      },
-    },
-    {
-      content: 'CE',
-      func: () => setInput(null),
-    },
-  ]
+	const buttons: Button[] = [
+		{
+			content: 'x²',
+			func: () => {
+				if (square(ARITHMETIC_ARR, input)) setInput(input + '**2')
+			},
+		},
+		{
+			content: 'xʸ',
+			func: () => {
+				if (input === null) return false
 
-  function normalOperation(event: MouseEvent) {
-    const elem = event.target as HTMLButtonElement
-    const lastPressedButton = elem.textContent!
+				if (power(ARITHMETIC_ARR, input, NUMBER_ARR)) setInput(input + '**')
+			},
+		},
+		{
+			content: '←',
+			func: () => {
+				if (input === null) return
 
-    if (input === null) return setInput(lastPressedButton)
+				if (input.endsWith('**2')) {
```

Comment:
Maybe there is a better way to signify squaring something, perhaps a special character replaces "**2" in the final output.